user/installing-dependencies.md

```diff
@@ -59,17 +45,109 @@ This example adds the APT repository for Varnish 3.0 for Ubuntu 12.04 to the loc
+### Installing Packages without an APT Repository
 
 For some projects, there may be a Debian/Ubuntu package available, but no corresponding APT repository. These are still easy to install, but require the extra step of downloading.
 
+If you're installing packages this way, make sure you download the correct version for your environment, either 12.04 LTS for the standard environment or 14.04 LTS for the Trusty environment.
+
 Say your project requires the pngquant tool to compress PNG files, here's how to download and install the .deb file:
 
     before_install:
       - wget http://pngquant.org/pngquant_1.7.1-1_i386.deb
       - sudo dpkg -i pngquant_1.7.1-1_i386.deb
 
-If you're installing packages this way, make sure they're available for Ubuntu 12.04, our current Linux platform.
+### Installing Packages with the APT Addon
+
+You can also use the APT addon.
+
+#### Adding APT Sources
+
+To add APT sources from the [source whitelist](https://github.com/travis-ci/apt-source-whitelist) before your custom build steps, use the `addons.apt.sources` key:
+
+``` yaml
+addons:
+  apt:
+    sources:
+    - deadsnakes
+    - ubuntu-toolchain-r-test
+```
+
+#### Adding APT Packages
+
+To install packages from the [package whitelist](https://github.com/travis-ci/apt-package-whitelist)  before your custom build steps, use the `addons.apt.packages` key:
+
+``` yaml
+addons:
+  apt:
+    packages:
+    - cmake
+    - time
+```
+
+> Note: When using APT sources and packages together, you need to make
+> sure they are under the same key space in the YAML file. e.g.
+
+``` yaml
+addons:
+ apt:
+   sources:
+   - ubuntu-toolchain-r-test
+   packages:
+   - gcc-4.8
+   - g++-4.8
+```
+
+## Installing Packages on Container Based Infrastructure
+
+To install packages not included in the default [container-based-infrastructure](/user/workers/container-based-infrastructure) you need to use the APT addon, as sudo apt-get is not available.
+
+### Adding APT Sources
+
+To add APT sources from the [source whitelist](https://github.com/travis-ci/apt-source-whitelist) before your custom build steps, use the `addons.apt.sources` key:
+
+``` yaml
+addons:
+  apt:
+    sources:
+    - deadsnakes
+    - ubuntu-toolchain-r-test
+```
+
+### Adding APT Packages
+
+To install packages from the [package whitelist](https://github.com/travis-ci/apt-package-whitelist)  before your custom build steps, use the `addons.apt.packages` key:
+
+``` yaml
+addons:
+  apt:
+    packages:
+    - cmake
+    - time
+```
+
+> Note: When using APT sources and packages together, you need to make
+> sure they are under the same key space in the YAML file. e.g.
+
+``` yaml
+addons:
+ apt:
+   sources:
+   - ubuntu-toolchain-r-test
+   packages:
+   - gcc-4.8
+   - g++-4.8
+```
+
+## Installing Packages on OSX
+
+To install packages that are not included in the [default OSX environment](/user/osx-ci-environment/) use [Homebrew](http://brew.sh) in your `.travis.yml`. For example, to install beanstalk:
```

Comment:
I'd link to https://docs.travis-ci.com/user/osx-ci-environment/#Compilers-%26-Build-toolchain instead as it's the one showing the packages installed?